### ISSUE TITLE
Implement eth_newFilter and eth_newPendingTransactionFilter

### ISSUE THREAD

**Issue by flexsurfer** [status-im/status-go#1110]
# Problem

https://github.com/status-im/status-react/issues/5162 

# Implementation

Implement `eth_newFilter` and `eth_newPendingTransactionFilter`.

# Background

Infura used to offer these methods (which was a questionable approach, but very convenient to us).
Now they remove support for all the filters whatsoever, so when we use an RPC mode, we need to re-implement these APIs by ourselves.

We can't just take whatever go-ethereum is offering either, because the blockchain service (ETH or LES) isn't running on a local node at that time, so we don't really have a list of pending transactions.

The proposed solution here is to implement an artificial list of "pending" transactions from the time when they are successfully sent to Infura to the time when the filter is being read.

**Comment by divan** [comment]
@flexsurfer can you provide more descriptive explanation of that? Starting from why we don't have it implemented at the moment and to the acceptance criteria and examples how to test it locally?
The more descriptive PR is the higher chances someone would jump in to fix it asap.

**Comment by divan** [comment]
Related: https://github.com/status-im/status-go/pull/1113

**Comment by mandrigin** [comment]
@divan added more background on the issue.

**Comment by sebastiandelgado** [comment]
@divan @flexsurfer #1113 just closed with the implementation for `newPendingTransactionFilter` in it. Let me know if there's any trouble with that interface, or any next steps.

**Comment by dshulyak** [comment]
I took a look at NewFilter implementation. and i think the easiest way is to implement it by smartly polling `eth_getLogs`. Which is still exposed by infura. And supports pretty much the same filtering criteria as `eth_newFilter` call.

There are few important things that catches my eye. In original implementation filters are closed unless polled regularly. We had similar problem with whisper. And it had very bad impact. But i think it is important to keep the same condition in our implementation. Cause unlike whisper methods, NewFilter will be used by dapps.

Original implementation notifies clients that certain logs were removed in case of the chain reorg. But since we are using multiple different ethereum nodes (because of load balancer), that are not perfectly synced between each other, we may miss part of notifications. If we query node A and it has block 0x01 at position 1 and that block has logs 1 and 2. Next time we query node B and it has block 0x11 at position 1 with different transactions and different logs, e.g. 1 and 3. In this situation we will never receive notification that log 2 was removed due to re-org.

If no one yet started working on filter for log I can implement it this week.

btw i didn't click unassign. for some reason github did it.
